tests/unit_tests/rng/device/moments/moments.cpp

```diff
@@ -961,6 +1049,190 @@ INSTANTIATE_TEST_SUITE_P(Philox4x32x10ExponentialIcdfAccDeviceMomentsTestsSuite,
                          Philox4x32x10ExponentialIcdfAccDeviceMomentsTests,
                          ::testing::ValuesIn(devices), ::DeviceNamePrint());
 
+class Philox4x32x10BetaCjaDeviceMomentsTests
+        : public ::testing::TestWithParam<sycl::device*> {};
+
+class Philox4x32x10BetaCjaAccDeviceMomentsTests
+        : public ::testing::TestWithParam<sycl::device*> {};
+
+// implementation uses double precision for accuracy
+TEST_P(Philox4x32x10BetaCjaDeviceMomentsTests, RealDoublePrecision) {
+    CHECK_DOUBLE_ON_DEVICE(GetParam());
+
+    rng_device_test<moments_test<oneapi::mkl::rng::device::philox4x32x10<1>,
```

Comment:
I propose to split float and double, if we can still test float on HW without FP64 support